src/python_testing/TC_IDM_4_2.py

```diff
@@ -0,0 +1,170 @@
+        return await self.read_single_attribute_check_success(
+            dev_ctrl=th, endpoint=0, cluster=cluster, attribute=attribute)
+
+    @async_test_body
+    async def test_TC_IDM_4_2(self):
+        
+        SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT = 0
+        
+        # Controller 1 Setup
+        CR1 = self.default_controller
+        CR1_node_id = self.matter_test_config.controller_node_id
+        
+        CR1_ace_full_access = Clusters.AccessControl.Structs.AccessControlEntryStruct(
+            privilege=Clusters.AccessControl.Enums.AccessControlEntryPrivilegeEnum.kAdminister,
+            authMode=Clusters.AccessControl.Enums.AccessControlEntryAuthModeEnum.kCase,
+            subjects=[CR1_node_id],
+            targets=[])
+        
+        acl = [CR1_ace_full_access]
+        await self.write_acl(acl)
+        
+        # Read ServerList attribute
+        self.print_step("0a", "CR1 reads the Descriptor cluster ServerList attribute from EP0")
+        ep0_servers = await self.read_descriptor_server_list_expect_success(CR1)
+        
+        # Check if ep0_servers contains the ICD Management cluster ID (0x0046)
+        if Clusters.IcdManagement.id in ep0_servers:
+            # Read the IdleModeDuration attribute from the DUT
+            self.print_step("0b", "CR1 reads from the DUT the IdleModeDuration attribute and sets SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT = IdleModeDuration")
+            logging.info("debux - Setting IcdManagement cluster...")
+            cluster = Clusters.Objects.IcdManagement
+            
+            logging.info("debux - Setting IdleModeDuration attribute...")
+            idle_mode_duration_attr = Clusters.IcdManagement.Attributes.IdleModeDuration
+        
+            logging.info("debux - Getting idleModeDuration attribute value...")
+            idleModeDuration = await self.read_single_attribute_check_success(
+                cluster=cluster, 
+                attribute=idle_mode_duration_attr, 
+                dev_ctrl=CR1, 
+                node_id=self.dut_node_id, 
+                endpoint=0)
+            
+            logging.info('debux - idleModeDuration: ' + idleModeDuration)
+            
+            SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT = idleModeDuration
+        else:
+            # Defaulting SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT to 60 minutes
+            self.print_step("0b", "Set SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT = 60 mins")
+            SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT = 3600
+        
+        # Step 1
+        self.print_step(1, "CR1 sends a subscription message to the DUT with MaxIntervalCeiling set to a value greater than SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT. DUT sends a report data action to the TH. CR1 sends a success status response to the DUT. DUT sends a Subscribe Response Message to the CR1 to activate the subscription.")
+        min_interval_floor_sec = SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT - 60
```

Comment:
This is dangerous - SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT could theoretically be less than 60. Use a min here.

Reply:
Unresolved: I see no change fixing the floor being 0 or negative. Needs a comment (and assert!) if this can never happen, otherwise use some form of setting on the floor, like:

```python
min_interval_floor_sec = max(1, SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT - 60)
```

Reply:
Updated